### ISSUE TITLE
A few questions about the predicted segmentation label generated by OAA.

### ISSUE THREAD

**Issue by justchenhao** [PengtaoJiang/OAA#5]
In your paper, in Table 2. "Comparisons of mIoU scores on the PASCAL VOC 2012 validation set when using different settings", I found that the mIoU(58.6%) of  OAA with MAX is higher than the mIoU(51.2%) of OAA+ with MAX and MCE. 
I have questions as follows:
- how to generate segmentation labels by OAA(58.6%).
- have you tried to use these proxy segmentation labels(58.6%) to train the segmentation neural networks.

**Comment by PengtaoJiang** [comment]
Hi, justchenhao. 
The code for generating segmentation labels is released in [link](https://drive.google.com/file/d/1SHQQBLZ_rarEB54tfrYJ0JVhku5a82EU/view?usp=drive_open) 
Please follow the instruction in README.

**Comment by GuoleiSun** [comment]
Hi Peng-Tao,
Based on your explanation, after we generate attention maps from the integral attention model by:
cd examples/oaa/exp2/
python eval.py 30000 0

if we want to generate segmentation labels, we need to use this code:  [link](https://drive.google.com/file/d/1SHQQBLZ_rarEB54tfrYJ0JVhku5a82EU/view?usp=drive_open)

**Comment by PengtaoJiang** [comment]
@GuoleiSun Hello, I cannot find your problem. ^_^

**Comment by GuoleiSun** [comment]
Hi Pent-Tao, I tried you code in [link](https://drive.google.com/file/d/1SHQQBLZ_rarEB54tfrYJ0JVhku5a82EU/view?usp=drive_open), but I am confused. The code needs sal_path and att_path, but we only have att_path generated by "python eval.py 30000 0". Right?

# set path for data
data_path = './'
train_lst_path = data_path + 'train_cls.txt'
im_path = data_path + 'JPEGImages/'
sal_path = data_path + 'saliency_aug/'
att_path = data_path + 'memory/'
save_path = './proxy-gt/'

What is sal_path and how to generate it? Also for you pytorch code of OAA, the same problem exists, it is not mentioned that how to generate segmentation labels in OAA-pytorch.

**Comment by GuoleiSun** [comment]
Hi Peng-Tao,
I tried your code to generate segmentation label [link](https://drive.google.com/file/d/1SHQQBLZ_rarEB54tfrYJ0JVhku5a82EU/view?usp=drive_open), but I am confused. The code needs "sal_path" and "att_path", but from my understanding, only "att_path" is available, right?

sal_path = data_path + 'saliency_aug/'
att_path = data_path + 'memory/'

Basically, how to use this code? Also, in OAA-pytorch, the same problem exists, it is not mentioned how to generate segmentation labels (proxy labels) ?  Thanks

**Comment by PengtaoJiang** [comment]
In Pre-computed Results Section, you can find the maps and scripts.

发件人：Guolei Sun <notifications@github.com>
发送日期：2019-12-03 22:35:02
收件人：PengtaoJiang/OAA <OAA@noreply.github.com>
抄送人：Jiangpengtao <pt.jiang@mail.nankai.edu.cn>,Comment <comment@noreply.github.com>
主题：Re: [PengtaoJiang/OAA] A few questions about the predicted segmentation label generated by OAA. (#5)
Hi Peng-Tao,
 I tried your code to generate segmentation label link, but I am confused. The code needs "sal_path" and "att_path", but from my understanding, only "att_path" is available, right?
sal_path = data_path + 'saliency_aug/'
 att_path = data_path + 'memory/'
Basically, how to use this code? Also, in OAA-pytorch, the same problem exists, it is not mentioned how to generate segmentation labels (proxy labels) ? Thanks

**Comment by GuoleiSun** [comment]
Hi, I downloaded everything in pre-computed results section, but I can not find the **script** to **generate** saliency maps. My question is: how do you generate the saliency maps, which are used to generate proxy labels.

And in your paper, you mentioned that you used the attention maps to get saliency maps. But the reference is **missing**. Please see the following screenshot from your paper. Could you correct the small mistake? Thanks a lot
![image](https://user-images.githubusercontent.com/22496918/70075404-069bbb80-15fd-11ea-9d89-2df1d43cfceb.png)

**Comment by PengtaoJiang** [comment]
We use the salinecy method [DSS](https://github.com/Andrew-Qibin/DSS) to generate maps.  We have correct this error and the latest version of our paper can be downloaded at the [cvpr open access](http://openaccess.thecvf.com/content_ICCV_2019/papers/Jiang_Integral_Object_Mining_via_Online_Attention_Accumulation_ICCV_2019_paper.pdf).

**Comment by GuoleiSun** [comment]
Hi Peng-Tao, could you share the scripts **directly** you used to generate saliency maps from attention maps? That's to say, to reproduce your results, wouldn't it be better if you can share the scripts which directly **takes attention maps as input** and **output saliency maps**, rather than referring  a [github project](https://github.com/Andrew-Qibin/DSS) which may take us days to figure out how you use it to generate saliency maps? Thanks and look forward to your reply

**Comment by GuoleiSun** [comment]
Hi Peng-Tao, could you share the scripts **directly** you used to generate saliency maps from attention maps? That's to say, to reproduce your results, wouldn't it be better if you can share the scripts which **directly takes attention maps as input and output saliency maps**, rather than referring a [DSS](https://github.com/Andrew-Qibin/DSS) which may take us days to figure out how you use it to generate saliency maps? Thanks and look forward to your reply
